vm/systemSmartContracts/auction.go

```diff
@@ -49,35 +49,31 @@ type stakingAuctionSC struct {
 	mutExecution          sync.RWMutex
 }
 
-// ArgsStakingAuctionSmartContract is the arguments structure to create a new StakingAuctionSmartContract
-type ArgsStakingAuctionSmartContract struct {
+// ArgsValidatorSmartContract is the arguments structure to create a new ValidatorSmartContract
+type ArgsValidatorSmartContract struct {
 	StakingSCConfig    config.StakingSystemSCConfig
 	GenesisTotalSupply *big.Int
-	NumOfNodesToSelect uint64
 	Eei                vm.SystemEI
 	SigVerifier        vm.MessageSignVerifier
 	StakingSCAddress   []byte
-	AuctionSCAddress   []byte
+	ValidatorSCAddress []byte
 	GasCost            vm.GasCost
 	Marshalizer        marshal.Marshalizer
 	EpochNotifier      vm.EpochNotifier
 }
 
-// NewStakingAuctionSmartContract creates an auction smart contract
-func NewStakingAuctionSmartContract(
-	args ArgsStakingAuctionSmartContract,
-) (*stakingAuctionSC, error) {
+// NewValidatorSmartContract creates an validator smart contract
+func NewValidatorSmartContract(
+	args ArgsValidatorSmartContract,
+) (*validatorSC, error) {
 	if check.IfNil(args.Eei) {
 		return nil, vm.ErrNilSystemEnvironmentInterface
 	}
 	if len(args.StakingSCAddress) == 0 {
 		return nil, vm.ErrNilStakingSmartContractAddress
 	}
-	if len(args.AuctionSCAddress) == 0 {
-		return nil, vm.ErrNilAuctionSmartContractAddress
-	}
-	if args.NumOfNodesToSelect < 1 {
-		return nil, fmt.Errorf("%w, value is %v", vm.ErrInvalidMinNumberOfNodes, args.NumOfNodesToSelect)
+	if len(args.ValidatorSCAddress) == 0 {
```

Comment:
is the check just for 0 enough? What happens if the address is not 32 bytes long? Should there also be a check with the MessageSignVerifier internals, should these be always be synchronized in terms of keys and signatures lengths?

Reply:
those signatures exist. the input for this arguments are comming from a set of constants, used throughout the project. if one is modified with wrong length a lot of tests will fail.

Reply:
ok